app/services/user_subscriptions/create.rb

```diff
@@ -0,0 +1,72 @@
+module UserSubscriptions
+  class Create
+    attr_accessor :user, :source_type, :source_id, :subscriber_email, :response
+
+    def self.call(*args)
+      new(*args).call
+    end
+
+    def initialize(user, user_subscription_params)
+      @user = user
+      @source_type = user_subscription_params[:source_type]
+      @source_id = user_subscription_params[:source_id]
+      @response = new_response
```

Comment:
Suggestion:

1. Assign the response struct to a constant (like Ruby does for all classes):
    ```ruby
    Response = Struct.new(:success, :data, :error, keyword_init: true)
    ```
3. Change this to:
    ```ruby
    @response = Response.new(success: false)
    ```
    You can leave off the arguments that are `nil`, structs are smart about this:
    ```ruby
    Response = Struct.new(:success, :data, :error, keyword_init: true)
    Response.new(success: false)
    #=> #<struct Response success=false, data=nil, error=nil>
    ```

If we start using this pattern in more places, we can actually extract a `BaseService` for others to inherit that already includes things like the response struct (and I usually add a method `success?` to wrap the instance variable in a more idiomatic way, etc).

Reply:
1. I tried that originally, you get `SyntaxError:  dynamic constant assignment`.

Reply:
JK, looks like I need to just do a little more refactoring to get it to work with the constant. Updated.